orquesta/specs/native/v1/models.py

```diff
@@ -156,6 +156,8 @@ def has_retry(self):
     def render(self, in_ctx):
         action_specs = []
 
+        item_ctx_value = ctx_util.copy_context(in_ctx)
```

Comment:
having the copy here will slow down non with items .  move the copy in the else clause.